src/Symfony/Component/Form/Exception/TransformationFailedException.php

```diff
@@ -18,4 +18,35 @@
  */
 class TransformationFailedException extends RuntimeException
 {
+    private $messageKey;
+    private $messageData;
+
+    public function __construct($message = '', $code = 0, \Exception $previous = null, $messageKey = null, $messageData = [])
+    {
+        parent::__construct($message, $code, $previous);
+
+        $this->setSafeMessage($messageKey, $messageData);
+    }
+
+    /**
+     * Set a message that will be shown to the user.
+     *
+     * @param string $messageKey  The message or message key
+     * @param array  $messageData Data to be passed into the translator
+     */
+    public function setSafeMessage($messageKey, array $messageData = array())
```

Comment:
Another elegant solution would be to use a static constructor, which would allow to take the `$messageKey` and `$messageData` as first arguments. But I didn't regarding https://github.com/symfony/symfony/pull/15882#r40289250